src/test/java/com/training/web/tests/flipkart/FlipkartTest.java

```diff
@@ -67,4 +68,80 @@ public void test() throws WebDriverException {
          */
         driver.close();
     }
+
+    @Test
+    public void testCartAddition() {
+
+        Integer[] products = {2, 3};
+        Integer[] size = {8, 8};
+
+        /**
+         * System Property for Chrome Driver
+         */
+        System.setProperty("webdriver.chrome.driver", "package/resources/chromedriver.exe");
+
+        /**
+         * Instantiate a ChromeDriver class.
+         */
+        ChromeOptions options = new ChromeOptions().addArguments("incognito");
+        DesiredCapabilities capabilities = DesiredCapabilities.chrome();
+        capabilities.setCapability(ChromeOptions.CAPABILITY, options);
+        WebDriver driver = new ChromeDriver(capabilities);
+
+        /**
+         * Launch Website and maximise
+         */
+        driver.navigate().to("https://www.flipkart.com/");
+        driver.manage().timeouts().implicitlyWait(10, TimeUnit.SECONDS);
+        driver.manage().window().maximize();
+        String parentWindow = driver.getWindowHandle();
+
+        /**
+         * Close popup and search for shoes
+         */
+        ResultPage resultPage = new FlipkartHomePage(driver).closePopup().sendKeysToSearchBox("shoes")
+                .clickSearch().sortLowToHigh();
+
+        /**
+         * Select and add to cart item 2 and 3
+         */
+        ArrayList<String> productNames = new ArrayList<>();
+        ArrayList<Integer> priceList = new ArrayList<>();
+        resultPage.getProductsList();
+
+        for (int i : products) {
+            ProductPage productPage = resultPage.clickProduct(i).clickSize("7");
+            productNames.add(productPage.getProductName());
+            priceList.add(productPage.getProductPrice());
+            productPage.addToCart();
+            driver.close();
+            driver.switchTo().window(parentWindow);
+        }
+        CartPage cartPage = resultPage.goToCart(driver);
+
+        /**
+         * Asserting for right products
+         */
+        ArrayList<String> cartProducts = cartPage.getProductNames();
+        int count = 0;
+        for (String cartProductName : cartProducts) {
+            for (String productListName : productNames) {
+                if (productListName.contains(cartProductName)) {
+                    count = 1;
+                }
+            }
+            Assertions.assertEquals(count, 1, "Product not in cart");
+        }
```

Comment:
do we need 2 for loops if we play with the index - as the size of both cartProducts & productNames are same?

Reply:
@akhil7000 please add this to the pending discussion topics list.